tests/shapes/test_tpms.py

```diff
@@ -575,26 +575,22 @@ def test_infill_bounds_match_obj_bounds(kwarg: dict[str, int | float]) -> None:
     assert np.all(grid_dim < obj_dim + tpms.cell_size)
 
 
-def test_infill_fills_the_object_with_any_normal_orientation() -> None:
-    """Test if the object is filled correctly with any normal orientation."""
-    mesh = microgen.Box(dim=(1.0, 1.0, 1.0)).generate_vtk()
-    first = microgen.Infill(
-        obj=mesh,
+def test_infill_cylinder_has_expected_volume() -> None:
+    """Test if an infilled cylinder has the expected volume."""
+    density = 0.5
+    cylinder = microgen.Cylinder().generate_vtk()
+    infill = microgen.Infill(
+        cylinder,
         surface_function=microgen.surface_functions.gyroid,
-        offset=TEST_DEFAULT_OFFSET,
-        cell_size=0.5,
-    )
-    mesh = mesh.triangulate()
-    mesh.flip_normals()
-    second = microgen.Infill(
-        obj=mesh,
-        surface_function=microgen.surface_functions.gyroid,
-        offset=TEST_DEFAULT_OFFSET,
-        cell_size=0.5,
+        density=density,
+        repeat_cell=2,
     )
 
-    assert np.allclose(first.sheet.volume, second.sheet.volume)
-    assert first.sheet.points.shape == second.sheet.points.shape
+    components = infill.sheet.connectivity().point_data["RegionId"]
+    n_unique = len(np.unique(components))
+
+    assert n_unique == 1
```

Comment:
I think this assert should be placed in a different test.

By the name of this test `test_infill_cylinder_has_expected_volume` you are checking the expected **behavior** of obtaining an expected volume value after the infill operation.

However this assert in line 592, check the **behavior** of obtaining only 1 connected component after performing an infill operation. This is a different **behavior** that should be tested in a different test independently, you can split this test as follows:

`test_infill_cylinder_has_expected_volume`
`test_infill_cylinder_returns_single_connected_component_mesh`

Reply:
Done, thanks!